packages/custom-esbuild/src/dev-server/index.ts

```diff
@@ -0,0 +1,79 @@
+import * as path from 'node:path';
+import { BuilderContext, createBuilder, targetFromTargetString } from '@angular-devkit/architect';
+import {
+  DevServerBuilderOptions,
+  DevServerBuilderOutput,
+  executeDevServerBuilder,
+} from '@angular-devkit/build-angular';
+import { IndexHtmlTransform } from '@angular-devkit/build-angular/src/utils/index-file/index-html-generator';
+import { getSystemPath, json, normalize } from '@angular-devkit/core';
+import { Observable, from, switchMap } from 'rxjs';
+import type { Plugin } from 'esbuild';
+import type { Connect } from 'vite';
+
+import { loadModule } from '../utils';
+import { patchBuilderContext } from './patch-builder-context';
+import {
+  CustomEsbuildApplicationSchema,
+  CustomEsbuildDevServerSchema,
+} from '../custom-esbuild-schema';
+
+export function executeCustomDevServerBuilder(
+  options: CustomEsbuildDevServerSchema,
+  context: BuilderContext
+): Observable<DevServerBuilderOutput> {
+  const buildTarget = targetFromTargetString(
+    // `browserTarget` has been deprecated.
+    options.buildTarget ?? options.browserTarget!
+  );
+
+  async function getBuildTargetOptions() {
+    return (await context.getTargetOptions(
```

Comment:
Guess this can be simplified now

Reply:
getBuildTargetOptions is not needed as a separate function, getTargetOptions can be used directly in "from".

Reply:
Its return type is `type|Promise`. I used the separate function explicitly to await the value (and also to cast it).

Reply:
OK, makes it more clean, ofc with casting. 
Didn't know there is a difference between returning the promise and awaiting it in a separate function (will put in on my reading list).